### ISSUE TITLE
Fatal error (crash) when flattening fixed length arrays using the flatten() function

### ISSUE THREAD

**Issue by reinout91** [duckdb/duckdb#11760]
### What happens?

When I run in DuckDB in DBeaver, driver version: 0.10.2 and I do a flatten on fixed length arrays using the query below, DuckDB and DBeaver crash and shut down.

`SELECT flatten([[1, 2, 3]::integer[3], [4, 5, 6]::integer[3]]);`

Processor: 12th Gen Intel(R) Core(TM) i7-12700H   2.30 GHz
Installed RAM: 32,0 GB (31,7 GB usable)
64-bit operating system, x64-based processor

### To Reproduce

`SELECT flatten([[1,2,3]::integer[3],[4,5,6]::integer[3]]);`

In DuckDB version 0.10.2

### OS:

Windows

### DuckDB Version:

0.10.2

### DuckDB Client:

DBeaver

### Full Name:

Reinout Holtrup

### Affiliation:

Alliander

### What is the latest build you tested with? If possible, we recommend testing with the latest nightly build.

I have tested with a stable release

### Did you include all relevant data sets for reproducing the issue?

Yes

### Did you include all code required to reproduce the issue?

- [X] Yes, I have

### Did you include all relevant configuration (e.g., CPU architecture, Python version, Linux distribution) to reproduce the issue?

- [x] Yes, I have

**Comment by Tishj** [comment]
```
D SELECT flatten([[1, 2, 3]::integer[3], [4, 5, 6]::integer[3]]);
┌──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│ flatten(main.list_value(CAST(main.list_value(1, 2, 3) AS INTEGER[3]), CAST(main.list_value(4, 5, 6) AS INTEGER[3]))) │
│                                                       int32[]                                                        │
├──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│ [1, 2, 3, 4, 5, 6]                                                                                                   │
└──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
```
```
D select version();
┌────────────────┐
│   version()    │
│    varchar     │
├────────────────┤
│ v0.10.3-dev300 │
└────────────────┘
```

**Comment by reinout91** [comment]
Okay, seems to be a DBeaver issue in that case!

`
D SELECT flatten([[1, 2, 3]::integer[3], [4, 5, 6]::integer[3]]);
┌──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│ flatten(main.list_value(CAST(main.list_value(1, 2, 3) AS INTEGER[3]), CAST(main.list_value(4, 5, 6) AS INTEGER[3]))) │
│                                                       int32[]                                                        │
├──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│ [1, 2, 3, 4, 5, 6]                                                                                                   │
└──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
D select version();
┌───────────┐
│ version() │
│  varchar  │
├───────────┤
│ v0.10.2   │
└───────────┘
`

**Comment by Tishj** [comment]
It might be in the JDBC driver then ?

**Comment by Mause** [comment]
I'm able to replicate this with 0.10.0 via JDBC, but not with 0.10.2 - please double check the version you're using with `pragma version` @reinout91

**Comment by reinout91** [comment]
A, I saw that the JDBC driver used was still 0.10.1. It was fixed by upgrading to 0.10.2 manually.
Thank you!
![image](https://github.com/duckdb/duckdb/assets/10004995/3a47efb8-82c3-40cc-9c22-e78a72fbacae)

![image](https://github.com/duckdb/duckdb/assets/10004995/2932e946-dbe1-4611-8bfa-aa748952918a)
